homeassistant/components/synology_dsm/common.py

```diff
@@ -82,6 +83,31 @@ def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
         self._with_upgrade = True
         self._with_utilisation = True
 
+        self._login_future: asyncio.Future[None] | None = None
+
+    async def async_login(self) -> None:
```

Comment:
Since its very likely all the coordinators will try to login at once, I wrapped it in the future so only the first one does